### PR TITLE
Add statistics atom feed + email

### PR DESCRIPTION
**Pull request by alicebartlett**
Pivotal: https://www.pivotaltracker.com/n/projects/1261204/stories/87099378

I've taken @camilleldn's WIP branch
https://github.com/alphagov/whitehall/compare/statistics-feed?expand=1
and put it in here, along with support in the statistics controller.

So if you go to gov.uk/government/statistics.atom you'll get an atom feed.
I'm not sure I've gotten this right, it definitely works but I'm very unfamiliar
with whitehall and atom.

And if you go to  eg `government/email-signup/new?email_signup%5Bfeed%5D=http%3A%2F%2Fwww.dev.gov.uk%2Fgovernment%2Fstatistics.atom%3Fpublication_filter_option%3Dstatistics%26topics%255B%255D%3Dborders-and-immigration` you can sign up to email notifications

### CODE REVIEW

**Comment by alicebartlett** [comment]
(Sorry, just thinking aloud here: it looks like if there is an atom feed, then a link is shown on the page, which is pretty neat and means I don't need to do any frontend changes. It does mean, however, that this shouldn't be merged until the email end-point is set up. I'll go do that. In the meantime if you have any thoughts on this code feel free to leave 'em here too.)

**Comment by boffbowsh** [comment]
It looks ok to me, my only suggestion would be a quick smoke test that when you request the atom feed, it doesn't error.

**Comment by elliotcm** [comment]
You may also need to make changes here to get emails: https://github.com/alphagov/whitehall/blob/master/lib/whitehall/gov_uk_delivery/subscription_url_generator.rb

File: app/controllers/statistics_controller.rb [hunk 1]
```diff
@@ -14,6 +14,11 @@ def index
       format.json do
         render json: StatisticsFilterJsonPresenter.new(@filter, view_context, PublicationesquePresenter)
       end
+      format.atom do
+        documents = Publicationesque.published_with_eager_loading(@filter.documents.map(&:id))
+        @statistics = Whitehall::Decorators::CollectionDecorator.new(
+          documents.sort_by(&:public_timestamp).reverse, PublicationesquePresenter, view_context)
```
Comment (tommyp) [inline app/controllers/statistics_controller.rb]: Minor thing, but you might want to put the closing bracket on a new line and each argument on its own line for better readability. Like this:
```
Whitehall::Decorators::CollectionDecorator.new(
  documents.sort_by(&:public_timestamp).reverse,
  PublicationesquePresenter, 
  view_context
)
```

**Comment by alicebartlett** [comment]
After a few days of prodding at this, I'm not sure how to get it working.

The feed is there, with the results I expect, but the urls of the items in the atom feed are all `/publications/` not `/statistics/`. These urls redirect to `/statistics/` so technically this would work for the user, but they should be correct in the first place. 

A possible culprit is `PublicDocumentRoutesHelper::document_url` not working correctly for statistics.

This branch has some helpfully failing tests, so the next person who comes to have a look at it should focus on getting them working. Once that is done, as @elliotcm has pointed out above, there is probably some work to do on getting the emails to send, this work will be in: https://github.com/alphagov/whitehall/blob/master/lib/whitehall/gov_uk_delivery/subscription_url_generator.rb

**Pull request closed without merging**